File: Data Structures & Algorithms/binary-tree-right-side-view/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        if not root:
            return result
        result.append(root.val) # initial root added
        queue = deque([root])

        while queue:
            size = len(queue)
            rightmost = None # put None here
            for i in range(size):
                node = queue.popleft()
                if node.left:
                    rightmost = node.left # its node.left or node.right . Next row basically
                    queue.append(node.left)
                if node.right:
                    rightmost = node.right
                    queue.append(node.right)
        
            if rightmost: # if there is a right
                result.append(rightmost.val) # then append

        return result
        

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        if (root == null) return 0;
        return goodnode(root, root.val);
        
    }

    public int goodnode(TreeNode node, int value)
    {
        if(node == null)
            return 0;
        if(node.val >= value)
            return 1 + goodnode(node.left, node.val) + goodnode(node.right, node.val);
        else
            return goodnode(node.left, value) + goodnode(node.right, value);

    }


}
